Boltdir/site-modules/metrics/files/collect_metrics_files.rb

```diff
@@ -0,0 +1,377 @@
+    #   collector module.  This is how often it tars up the json files.
+    #
+    # @return [void]
+    #
+    # @example
+    #   initialize(start_epoch, end_epoch, metrics_dir, output_dir, tar_file_name,
+    #              poll_interval, archive_interval, verbose)
+    #
+    # rubocop:disable Metrics/ParameterLists
+    def initialize(start_epoch, end_epoch, metrics_dir, output_dir, tar_file_name,
+                   poll_interval, archive_interval, verbose)
+      @start_epoch = start_epoch
+      @end_epoch = end_epoch
+      @metrics_dir = metrics_dir
+      @output_dir = output_dir
+      @tar_file_name = tar_file_name
+      @poll_interval = poll_interval
+      @archive_interval = archive_interval
+      @verbose = verbose
+
+      @staging_dir = Dir.mktmpdir
+    end
+    # rubocop:enable Metrics/ParameterLists
+
+    # Process the metrics dir, blindly assume any non-exempted dir is a valid service dir
+    #
+    # @author Randell Pelak
+    #
+    # @return [void]
+    #
+    # @example
+    #   inspect_metrics_dir_for_service_dirs
+    def inspect_metrics_dir_for_service_dirs
+      metrics_dir_obj = Dir.open @metrics_dir
+      metrics_dir_obj.each do |path|
+        next if path =~ /^scripts$|^bin$|^\.$|^\.\.$/
+
+        absolute_path = "#{metrics_dir_obj.path}/#{path}"
+        service = path
+        next unless File.directory? absolute_path
+
+        inspect_service_dir_for_metrics_files(absolute_path, service)
+      end
+    end
+
+    # Process potential service dirs or subdirs
+    #
+    # Uses recursion but doesn't care which subdir the files come from, they all go in the
+    # same output service dir
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] dir_path Service dir or subdir
+    # @param [string] service name of the service the directory belongs to
+    #
+    # @return [void]
+    #
+    # @example
+    #   inspect_service_dir_for_metrics_files(dir_path, service)
+    def inspect_service_dir_for_metrics_files(dir_path, service)
+      service_dir = Dir.open dir_path
+      service_dir.each do |path|
+        next if path =~ /^\.$|^\.\.$/
+
+        absolute_path = "#{service_dir.path}/#{path}"
+        if File.directory?(absolute_path)
+          inspect_service_dir_for_metrics_files(absolute_path, service)
+          next
+        end
+        if path =~ /.json$/
+          consider_json_for_inclusion(absolute_path)
+        elsif path =~ /.tar.gz$/
+          consider_tarfile_contents_for_inclusion(absolute_path, service)
+        end
+      end
+    end
+
+    # Determines if a json file is from the target window
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] filename json filename
+    #
+    # @return [boolean] true if yes, false if no
+    #
+    # @example
+    #   json_file_from_target_window(filename)
+    def json_file_from_target_window?(filename)
+      time_str = filename.gsub(/.json$/, "")
```

Comment:
Rather than expecting just a filename and gsubbing it, you can isolate the filename without the suffix using `File.basename`.  The input should also be validated to be a json file.

```
raise ArgumentError, "Argument is not a json filename" unless filename.match(/\.json$/i)
time_str = File.basename(filename, ".json")
```

Reply:
That raise caught a bug that tar files where getting passed to it.  Fixing that bug also sped it up a lot.